cassandra/cluster.py

```diff
@@ -3612,6 +3608,7 @@ def _set_result(self, host, connection, pool, response):
                 # we got some other kind of response message
                 msg = "Got unexpected message: %r" % (response,)
                 exc = ConnectionException(msg, host)
+                self._cancel_timer()
```

Comment:
I was wondering if this line is really necessary,  as the timer will be canceled inside inside
`self._set_final_exception(exc)`